dynawo/sources/Modeler/Common/DYNConnector.cpp

```diff
@@ -142,21 +142,42 @@ ConnectorContainer::mergeYConnector() {
   yConnectors_.assign(yConnectorsList.begin(), yConnectorsList.end());
 }
 
+
+int
+ConnectorContainer::getConnectorVarNum(const shared_ptr<SubModel>& subModel, const shared_ptr<Variable>& variable, bool flowConnector) {
+  int numVar;
+  if (flowConnector && variable->isAlias()) {
+    string id = subModel->name()+"_"+variable->getName();
+    boost::unordered_map<std::string, int>::const_iterator aliasIt = flowAliasNameToFictitiousVarNum_.find(id);
+    if (aliasIt != flowAliasNameToFictitiousVarNum_.end()) {
+      numVar = aliasIt->second;
+    } else {
+      numVar = sizeY_ + flowAliasNameToFictitiousVarNum_.size();
+      flowAliasNameToFictitiousVarNum_[id] = numVar;
+    }
+  } else {
+    numVar = subModel->getVariableIndexGlobal(variable);
+  }
+  return numVar;
+}
 void
 ConnectorContainer::mergeFlowConnector() {
   // order flow connectors
   flowConnectors_.clear();
   flowConnectorByVarNum_.clear();
+  flowAliasNameToFictitiousVarNum_.clear();
+  const bool flowConnector = true;
   list<shared_ptr<Connector> > flowConnectorsList;
   for (unsigned int i = 0; i < flowConnectorsDeclared_.size(); ++i) {
     shared_ptr<Connector> flowc(new Connector(*flowConnectorsDeclared_[i]));
     bool merged = false;
     for (vector<connectedSubModel>::iterator it = flowc->connectedSubModels().begin();
         it != flowc->connectedSubModels().end();
         ++it) {
-      const int numVar = it->subModel()->getVariableIndexGlobal(it->variable());
+      const boost::shared_ptr<Variable>& variable = it->variable();
+      int numVar = getConnectorVarNum(it->subModel(), it->variable(), flowConnector);
```

Comment:
Either define variable and use it in the method - getConnectorVarNum(it->subModel(), variable, flowConnector) - or do not define it ;)

Reply:
and this is why review by another person is always useful :)